admin/resources/js/components/Table.tsx

```diff
@@ -23,7 +38,57 @@ function Table<T extends Record<string, unknown>>({
     setGlobalFilter,
     state,
     preGlobalFilteredRows,
-  } = useTable<T>({ columns, data }, useGlobalFilter, useSortBy);
+  } = useTable<T>(
+    {
+      columns,
+      data,
+      initialState: {
+        hiddenColumns: hiddenCols,
```

Comment:
Because it only seems to let me hide columns, not regenerate them. Ex: Setting setHiddenColumns([]) does not cause the previously hidden columns to disappear.

Though adding "setHiddneColumns['id']" causes all three columns to disappear.

Reply:
Try using initialState.hiddenColumns along with column.getToggleHiddenProps(). If that works to reveal initially hidden columns, it would be a nice way to set initial hidden state.